ivy/stateful/module.py

```diff
@@ -705,7 +704,6 @@ def build(
         ret
             True for successfully built a module.
         """
-        self._dev = ivy.default(device, self._dev)
```

Comment:
as mentioned above, we can continue having `_dev` in the class which would have a read-only property corresponding to it. Let's probably rename this to `_device` instead as that's the vocab across the rest of the framework and add a `device` property to `ivy.Module` to access `self._device`

Reply:
Thanks for the refactor, could you please also add the `ivy.Module.device` property to allow access to the `_device` private variable?